psl-core/src/main/java/org/linqs/psl/database/rdbms/RDBMSDatabase.java

```diff
@@ -388,7 +388,7 @@ private PreparedStatement getAtomUpsert(Connection connection, PredicateInfo pre
     }
 
     private PreparedStatement getAtomDelete(Connection connection, PredicateInfo predicate, Term[] arguments) {
-        PreparedStatement statement = predicate.createDeleteStatement(connection, writeID);
+        PreparedStatement statement = predicate.createDeleteStatement(connection, allPartitionIDs);
```

Comment:
Why the change in functionality?
Why not just target the write apparition by default and only use other partitions if told to?